tests/ttnn/unit_tests/operations/test_softmax.py

```diff
@@ -13,6 +13,98 @@
 from models.utility_functions import torch_random
 
 
+def run_softmax_stable_with_program_cache(device, batch_size, h, w):
+    torch.manual_seed(0)
+
+    torch_input_tensor = torch_random((batch_size, h, w), -1000, 1000, dtype=torch.bfloat16)
```

Comment:
these tests don't make sense. 

Please add specific tests to describe what specifically was fixed with stable version. Test like [-1000, 1000] will not work anyway, both in pytorch and ttnn. 

These kind of tests are main target of optimization/improvement: 
torch_input_tensor = torch.tensor([100.0, 101.0], dtype=torch.bfloat16)
torch_input_tensor = torch.tensor([100.0, 1000.0], dtype=torch.bfloat16)
torch_input_tensor = torch.tensor([-100.0, -101.0], dtype=torch.bfloat16)
torch_input_tensor = torch.tensor([-1000.0, -100.0], dtype=torch.bfloat16)

Reply:
I added these tests above, but I disagree about the test doesn't make sense, the stable version allow large values ranges pass, while previously not passing.

Reply:
My point about test that it didn't test what was intended to. 
To provide a better perspective here:
What's the probability of all random numbers be -X (where X is large enough that previously it didn't work, but now it works). Let's assume X = 100. So the probability to generate proper test ((1000 - 100) / 2000) ** (batch_size * h * w).  

min batch = 1
min h = 32
min w = 1024
```
@pytest.mark.parametrize("batch_size", [1, 8])
@pytest.mark.parametrize("h", [32, 128])
@pytest.mark.parametrize("w", [1024, 1500])
```


P(of proper test) = 0.45 ** (32 * 1024) < 0.5 ^ 32768.
To put it in perspective, the chance is still less than flipping a coin heads 32768 times in a row.